cotel/src/main/scala/day1/InverseCaptcha.scala

```diff
@@ -0,0 +1,21 @@
+package day1
+
+object InverseCaptcha {
+
+  def captcha(captcha: String): Int = {
+    val parsedCaptcha = captcha.toList.map(_.asDigit)
+
+    if (!hasDuplicatedElements(parsedCaptcha)) 0
+    else recursiveCaptcha(parsedCaptcha :+ parsedCaptcha.head)
+  }
+
+  private def hasDuplicatedElements(captcha: List[Int]): Boolean =
```

Comment:
Isn't this method useless?  I mean, is not necessary as the "recursiveCaptcha" can do the same job implicitly and return 0 if there are no duplicated elements.

Reply:
Do you mean checking if there are no duplicates in each iteration? I suppose it could be an exit condition for the recursion and it would improve times.

On the other hand, I think it would be breaking Single Reponsability Principle because it would be performing more actions than just the algorithm which is pattern matching.

Really nice detail 👍

Reply:
I was more on the point of no worriying about exitpoints or duplicates at all. You run the method "recursiveCaptcha" either there are or there are not duplicates, if there are not, would return 0 without problem.

Reply:
Agree with @JoseLlorensRipolles